assignment/models/homework.py

```diff
@@ -48,6 +48,17 @@ def check_date(self):
     def onchange_subject_standard(self):
         self.subject_standard_assignment = self.standard_id.standard_id.id
 
+    @api.onchange('standard_id', 'teacher_id')
+    def onchange_stamdard_id(self):
+        domain = []
+        if self.teacher_id:
+            domain.append(
+                    ('teacher_ids', '=', self.teacher_id.id))
+        if self.standard_id:
+            domain.append(
+                ('standard_ids', '=', self.standard_id.standard_id.id))
+        return {'domain': {'subject_id': domain}}
+
```

Comment:
@AnuKPatel if the teacher_ids and standard_ids fields are x2Many, domain should have 'in' instead of '='.